### PR TITLE
feat(conversion): implement tool-specific JSON-to-Markdown converters

### PR DESCRIPTION
**Pull request by batradev**
## Description

- Develop converters for Quiz, Flashcards, Worksheet, and Syllabus tools
- Update Firestore structure to store original JSON and converted Markdown
- Initialize editorState.editHistory with initial version
- Handle edge cases like empty responses and special character escaping
- Ensure no back-conversion from Markdown to JSON
- Integrate converters into the tool generation flow
- Test Markdown generation for all tool types

## Related Issue

https://sharing.clickup.com/9018811771/v/6-901805122349-1/t/h/86er92b3a/231403108364eac

## Type of Change

Please select the type(s) of change that apply and delete those that do not. 

- [x] **New feature**: A non-breaking change that adds functionality.

## Proposed Solution

We implemented converters to transform JSON responses from various tools into Markdown for <DocumentEditor>. Updated Firestore to store both JSON and Markdown, added input validation, and designed the system to be modular for easier maintenance and future extensions.

## How to Test

NA

## Unit Tests

NA

## Documentation Updates

Indicate whether documentation needs to be updated due to this PR. 

- [ ] Yes
- [x] No

If yes, describe what documentation updates are needed and link to the relevant documentation.

## Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Additional Information

 Add any other information that might be useful for the reviewers.

### CODE REVIEW

**Comment by batradev** [comment]
# Quiz

# 1. Which planet is known as the "Red Planet"?
&nbsp;&nbsp;A. Mars
&nbsp;&nbsp;B. Venus
&nbsp;&nbsp;C. Jupiter
&nbsp;&nbsp;D. Saturn

# 2. Which of the following is NOT a planet in our Solar System?
&nbsp;&nbsp;A. Mercury
&nbsp;&nbsp;B. Jupiter
&nbsp;&nbsp;C. Saturn
&nbsp;&nbsp;D. Pluto

# 3. Which planet is the farthest from the Sun?
&nbsp;&nbsp;A. Jupiter
&nbsp;&nbsp;B. Saturn
&nbsp;&nbsp;C. Neptune
&nbsp;&nbsp;D. Uranus

# 4. Which of the following is NOT a planet in our solar system?
&nbsp;&nbsp;A. Mercury
&nbsp;&nbsp;B. Venus
&nbsp;&nbsp;C. Neptune
&nbsp;&nbsp;D. Pluto

# 5. Which planet is known as the "Red Planet"?
&nbsp;&nbsp;A. Mercury
&nbsp;&nbsp;B. Venus
&nbsp;&nbsp;C. Mars
&nbsp;&nbsp;D. Jupiter



# Flashcards

### Astronomy
The study of the universe beyond Earth.

---

### Solar System
A system centered around the Sun, consisting of planets, moons, asteroids, and comets.

---

### Planet
A celestial body that orbits a star.

---

### Star
A massive, hot ball of gas primarily composed of hydrogen and helium.


# Worksheet

# Question 1: Multiple Choice

Which planet is described as Earth's twin because of its similar size?

&nbsp;&nbsp;A. Mars
&nbsp;&nbsp;B. Venus
&nbsp;&nbsp;C. Mercury
&nbsp;&nbsp;D. Jupiter

**Answer:** B
**Explanation:** The text states that Venus is called Earth's twin because it's almost the same size as Earth.

# Question 2: True/False

True or False: Jupiter is the largest planet in our solar system and has a giant storm known as the Great Red Spot.

**Answer:** True
**Explanation:** The provided text states that Jupiter is the largest planet and has a giant storm called the Great Red Spot.  This makes the statement true.

# Question 3: True/False

According to the text, Saturn is known for its rings made of ice and rock.  True or False?

**Answer:** True
**Explanation:** The text explicitly states that "Saturn is famous for its beautiful rings made of ice and rock."

# Question 4: Fill in the Blank

The Sun, which is a ____, is at the center of our ____.  It provides ____ and ____ to the ____, including Earth.

**Word Bank:**
• star
• planets
• Solar System
• light
• warmth

**Answers:**
1. star
2. Solar System
3. light
4. warmth
5. planets

**Explanation:** The passage explains that the Sun, a star, is located at the center of our Solar System. It is the source of light and warmth for all the planets within the system, including our own planet Earth.

# Question 5: Fill in the Blank

The largest planet in our solar system is ____, which is known for its ____.  ____ is the closest planet to the sun, and is ____ during the day and ____ at night.

**Word Bank:**
• Jupiter
• Great Red Spot
• Mercury
• hot
• freezing

**Answers:**
1. Jupiter
2. Great Red Spot
3. Mercury
4. hot
5. freezing

**Explanation:** According to the text, Jupiter is the largest planet and has a giant storm called the Great Red Spot. Mercury is the smallest and closest to the Sun, experiencing extreme temperature differences between day and night.



# Syllabus

| Section | Details |
|---------|----------|
| Course Title | Introduction to Astronomy |
| Grade Level | 5th Grade |
| Description | This course provides a basic introduction to astronomy, the science of the universe. Students will learn about our solar system, including the Sun, the eight planets, and other celestial bodies like moons, asteroids, and comets.  The course will focus on the unique characteristics of each planet: Mercury (smallest and closest to the Sun), Venus (similar in size to Earth), Earth (supports life), Mars (dusty, red planet), Jupiter (largest with the Great Red Spot), Saturn (prominent rings), Uranus (rotates on its side), and Neptune (farthest and windiest).  Additionally, students will learn about stars, including our Sun, and how they generate light and heat through nuclear reactions. |
| Objectives | • Introduce students to the basic components of our solar system, including the Sun, planets, and Earth's moon.<br>• Explain the role of the Sun as a star and its relationship to the planets.<br>• Develop students' understanding of the Earth's rotation and its relationship to day and night.<br>• Foster an appreciation for the vastness of space and the unique characteristics of celestial bodies. |
| Learning Outcomes | • Students will be able to correctly identify the Sun as a star and Earth as a planet.<br>• Students will be able to name at least five of the eight planets in our solar system.<br>• Students will be able to explain that the Earth orbits the Sun and the moon orbits the Earth.<br>• Students will be able to describe the concept of day and night as a result of Earth's rotation.<br>• Students will be able to visually differentiate between the Sun, the Earth, and the moon in illustrations.<br>• Given a picture of our solar system, students will be able to point to the Sun and Earth with 80% accuracy.<br>• Given a verbal prompt, students will be able to explain that the Sun provides light and heat to Earth with 70% accuracy. |

**Comment by batradev** [comment]
The store contains all required data
store state

**Review by liamohkay — APPROVED** [review]
LGTM.

File: tools/data/slices/toolsSlice.js [hunk 1]
```diff
@@ -38,6 +42,18 @@ const tools = createSlice({
...
       state.response = action.payload;
     },
+    setMarkdownContent: (state, action) => {
+      state.editorState.markdownContent = action.payload;
+    },
+    addStateToEditHistory: (state, action) => {
+      state.editorState.editHistory = [
+        ...state.editorState.editHistory,
+        action.payload,
+      ];
```
Comment (liamohkay) [inline tools/data/slices/toolsSlice.js:52]:
``` "editHistory": [
      {
        "content": string,            // snapshot of Markdown
        "timestamp": timestamp,
        "type": "initial" | "auto-save" | "manual-save" | "restore"
      }
      // ... up to 15 total
    ]
    
   ``` 
   
   Match the save to this object from the 2.2 reqs.

**Review by liamohkay — CHANGES_REQUESTED** [review]
1 small change needed to match the 2.2 issue requirements. Otherwise looks great.

File: tools/data/slices/toolsSlice.js (hunk 1 above)
Comment (batradev) [reply tools/data/slices/toolsSlice.js:52]: Hey @liamohkay, I pushed the requested [changes](https://github.com/liamohkay/marvel-platform/compare/f46aae141c5e403f7347b9276b02dcde8adda760..093200d51f13808120f759e213b36e98371c2ce9)

**Review by liamohkay — APPROVED** [review]
Great work Yullia!
